cypress/integration/admin_portal/tests/test_code_management.spec.js

```diff
@@ -1,55 +1,163 @@
   const landingPage = new LandingPage()
   const codeManagementDashboard = new CodeManagementPage()
+  const coupons = new EnterpriseCoupons()
+  let couponId = null
+  let quantity
 
   before(function(){
     cy.login_using_api(Cypress.env('ADMIN_PORTAL_USER_EMAIL'), Cypress.env('ADMIN_PORTAL_USER_PASSWORD'))
+    const couponType = 'discount_single_course_single_use_percentage'
+    coupons.loginToEcommerce()
+    coupons.prepareCouponData(couponType).then((couponData) => {
+      coupons.createCoupon(couponData[couponType]).then((response) => {
+        this.couponId = response.body.coupon_id
+      })
+    })
+    coupons.fetchCouponReport(this.couponId).then((response) => {
+      // Coupon data from fixtures
+      this.coupon_report = response.body
+      this.couponName = this.coupon_report.match(/Test_Coupon_\w+/g)[0]
+      this.dates = this.coupon_report.match(/\w+ \d+, \d+/g)
+    })
+    cy.fixture('coupon_creation_data').then((couponData) => {
+      this.quantity = couponData['discount_single_course_single_use_percentage']['quantity']
+    })
   })
 
   beforeEach(function () {
     Cypress.Cookies.preserveOnce('edxloggedin', 'stage-edx-user-info', 'stage-edx-sessionid')
     cy.visit('/')
     landingPage.goToEnterprise('SuccessFactors')
-    landingPage.openCodeManagement()
+    landingPage.openCodeManagement() 
   })
 
   it('Verifies the validation checks on REQUEST MORE CODES form', function () {
-   codeManagementDashboard.requestMoreCodes()
-   codeManagementDashboard.getFormField('emailAddress').should('have.attr', 'value', 'zrana@edx.org')
-   codeManagementDashboard.getFormField('enterpriseName').should('have.attr', 'value', 'SuccessFactors')
-   codeManagementDashboard.getLabels('Company').should('have.text', 'Company*').children().should('have.class', 'required')
-   codeManagementDashboard.getLabels('Email').should('have.text', 'Email Address*').children().should('have.class', 'required')
-   codeManagementDashboard.getLabels('Codes').should('have.text', 'Number of Codes')
-   codeManagementDashboard.getRequestCodesButton().should('have.attr', 'type', 'submit').and('have.text', 'Request Codes')
-   codeManagementDashboard.getCancelButton().should('have.attr', 'href', '/successfactors/admin/coupons').and('have.text', 'Cancel')
-   codeManagementDashboard.getFormField('emailAddress').clear().prev().click()
-   codeManagementDashboard.getInvalidFeedback().should('have.text', 'This field is required.')
-   codeManagementDashboard.getFormField('emailAddress').type('zrana@')
-   codeManagementDashboard.getInvalidFeedback().should('have.text', 'Must be a valid email address.')
-   codeManagementDashboard.getRequestCodesButton().should('have.attr', 'disabled')
-  })
+    codeManagementDashboard.requestMoreCodes()
+    codeManagementDashboard.getFormField('emailAddress').should('have.attr', 'value', 'zrana@edx.org')
+    codeManagementDashboard.getFormField('enterpriseName').should('have.attr', 'value', 'SuccessFactors')
```

Comment:
avoid using hard coded values, you can fetch Successfactor from the environment variables in config file, the email should be fetched from there as well